### PR TITLE
proxy: shuffle endpoints

### PR DESCRIPTION
**Pull request by bakins**
Shuffle endpoints to avoid being "stuck" to a single cluster member.

### CODE REVIEW

File: Documentation/proxy.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ etcd can now run as a transparent proxy. Running etcd as a proxy allows for easi
 
 etcd currently supports two proxy modes: `readwrite` and `readonly`. The default mode is `readwrite`, which forwards both read and write requests to the etcd cluster. A `readonly` etcd proxy only forwards read requests to the etcd cluster, and returns `HTTP 501` to all write requests. 
 
+etcd will shuffle the list of cluster members periodically to avoid sending all connections to a single member.
```
Comment (xiang90) [inline Documentation/proxy.md]: To make it clear: `The proxy will`

File: proxy/director.go [hunk 2]
```diff
@@ -65,6 +66,13 @@ func (d *director) refresh() {
 		}
 		endpoints = append(endpoints, newEndpoint(*uu))
 	}
+
+	// shuffle array to avoid connections being "stuck" to a single endpoint
+	for i := range endpoints {
```
Comment (xiang90) [inline proxy/director.go]: a simple shuffling is good enough for now. If we want the proxy to coalesce watches, we need to be more careful with this.

**Comment by xiang90** [comment]
LGTM

File: proxy/director.go (hunk 2 above)
Comment (bakins) [inline proxy/director.go]: Correct. This is a quick fix for the short-term.  When we coalesce watches, we will need the watches to be "sticky" to a given endpoint - or at least all watches for a given key.

**Comment by xiang90** [comment]
@bakins Thanks!

**Pull request merged**